plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -963,6 +966,7 @@ public OptionalLong executeDelete(ConnectorSession session, ConnectorTableHandle
 
         icebergTable.newDelete()
                 .deleteFromRowFilter(toIcebergExpression(handle.getEnforcedPredicate()))
+                .set(TRINO_QUERY_ID_NAME, session.getQueryId())
```

Comment:
If I make the table partitioned, I can perform two appends to two different partitions, then perform a delete by partition query, and then finally an optimize and ensure all of those have the expected `trino_query_id`.

1) Is that too much for one test?
2) The materialized views presently have `PRESTO_QUERY_ID_NAME` in them. Should we add `trino_query_id` into the materialized views created from Iceberg tables so that Iceberg "tables" consistently have `trino_query_id` associated with them?

For materialized views, it's not in the `snapshots` table as there isn't a snapshots table (afaik) so that might be overkill.

Also, should we add `trino_version` (or `PRESTO_VERSION_NAME`) into the summary as well (or make a follow up ticket)?

Reply:
> If I make the table partitioned, I can perform two appends to two different partitions, then perform a delete by partition query, and then finally an optimize and ensure all of those have the expected trino_query_id.
> Is that too much for one test?

I think it's ok for now. Just keep in mind that all the places where you persist the Trino query ID changes should be tested.  BTW , table structure  changes ( add/drop column , table/column comments, TrinoHiveCatalog) operations are not supposed to persist a Trino query ID?

Reply:
> Also, should we add trino_version (or PRESTO_VERSION_NAME) into the summary as well (or make a follow up ticket)?

I find this information also good for debugging purposes in the context of rapid releases for Trino. 
I couldn't find any reference to the Trino version within the  JSON (over Trino admin web UI) for the query. 
Could you please file a new ticket to follow up later on this topic?

Reply:
> BTW , table structure changes ( add/drop column , table/column comments, TrinoHiveCatalog) operations are not supposed to persist a Trino query ID?

Reply:
I've added test cases for everything that adds a `PendingUpdate`. So even queries that result in 2+ snapshots, each snapshot will have the metadata.

Plus one end to end type test.

Working on integrating this into the newer catalog setup, but might do that in a separate PR (going to take a stab at it now though).